### ISSUE TITLE
ic-light was completely disabled when updating ComfyUI torch to 2.4.0, now it doesn’t work even in version 1.5 =(

### ISSUE THREAD

**Issue by ffdown** [kijai/ComfyUI-IC-Light#54]
ic-light was completely disabled when updating ComfyUI torch to 2.4.0, now it doesn’t work even in version 1.5 =(

```
Error occurred when executing LoadAndApplyICLightUnet:

IC-Light: Could not patch calculate_weight

File "X:\ComfyUI_windows_portable\ComfyUI\execution.py", line 317, in execute
output_data, output_ui, has_subgraph = get_output_data(obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "X:\ComfyUI_windows_portable\ComfyUI\execution.py", line 192, in get_output_data
return_values = _map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "X:\ComfyUI_windows_portable\ComfyUI\execution.py", line 169, in _map_node_over_list
process_inputs(input_dict, i)
File "X:\ComfyUI_windows_portable\ComfyUI\execution.py", line 158, in process_inputs
results.append(getattr(obj, func)(**inputs))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "X:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-IC-Light\nodes.py", line 83, in load
raise Exception("IC-Light: Could not patch calculate_weight")
```

**Comment by kijai** [comment]
Is that the whole error? It has nothing to do with torch version though, make sure these nodes are up to date as that error is caused by comfy moving a function, which I have fixed.

**Comment by ffdown** [comment]
> Is that the whole error? It has nothing to do with torch version though, make sure these nodes are up to date as that error is caused by comfy moving a function, which I have fixed.

I installed COMFYUI from scratch and downloaded the extension again, the error does not disappear.

**Comment by kijai** [comment]
All I know is that it works with both the node and ComfyUI being latest versions currently, as I specifically fixed it for that. I tried to make it backwards compatible as well, but haven't tested that.

**Comment by ffdown** [comment]
I apologize, my mistake was corrected this way. You never know, maybe someone didn’t understand what the problem was, but in fact ComfyUI was not updated for me via git pull.

checking existing branches
get branch -a
branch selection
git checkout master
Set up tracking for the local master branch:
git branch --set-upstream-to=origin/master master
Now run git pull:
git pull

P.S and of course the last question of the day, is it expected that ICLight for FLUX would be just a brilliant thing...

**Comment by kijai** [comment]
> I apologize, my mistake was corrected this way. You never know, maybe someone didn’t understand what the problem was, but in fact ComfyUI was not updated for me via git pull.
> 
> checking existing branches get branch -a branch selection git checkout master Set up tracking for the local master branch: git branch --set-upstream-to=origin/master master Now run git pull: git pull
> 
> P.S and of course the last question of the day, is it expected that ICLight for FLUX would be just a brilliant thing...

The model itself is only available for 1.5, so we are limited to that. You can still use it on Flux outputs of course, just probably drops the quality. There's a node called "Detail Transfer" that I added which will help mitigate the issue of lost sharpness that should help as well.

**Comment by ffdown** [comment]
> > I apologize, my mistake was corrected this way. You never know, maybe someone didn’t understand what the problem was, but in fact ComfyUI was not updated for me via git pull.
> > checking existing branches get branch -a branch selection git checkout master Set up tracking for the local master branch: git branch --set-upstream-to=origin/master master Now run git pull: git pull
> > P.S and of course the last question of the day, is it expected that ICLight for FLUX would be just a brilliant thing...
> 
> The model itself is only available for 1.5, so we are limited to that. You can still use it on Flux outputs of course, just probably drops the quality. There's a node called "Detail Transfer" that I added which will help mitigate the issue of lost sharpness that should help as well.

special thanks for Detail transfer I didn’t know about its existence, I should periodically update the examples in the folder, otherwise they tend to become outdated...
